src/cli.js

```diff
@@ -23,6 +24,16 @@ function getProcessEnvs(val) {
   return pEnv;
 }
 
+function checkInternet() {
+  dns.lookup('npmjs.com', (err) => {
+    if (err && err.code === 'ENOTFOUND') {
+      console.log('\nNo Internet Connection\nTo offline build, download electron from https://github.com/electron/electron/releases\nand place in ~/AppData/Local/electron/Cache/ on Windows,\n~/.cache/electron on Linux or ~/Library/Caches/electron/ on Mac\nUse --electron-version to specify the version you downloaded.');
+      process.exit(1);
```

Comment:
@mdh34 but `process.exit()`-ing here defeats the purpose, as it would make building offline altogether impossible, with or without cached files 🙂. You should

* Either do your `checkInternet` work later, when you determined user is offline AND Electron cached files are absent. Only when *both* of these conditions are met, `exit(1)` is the thing to do.
* Or leave the check where it is now (meaning it will always be done, which doesn't seem bad), but do not `exit()`, meaning packaging offline will be attempted (and maybe fail), but at least we'll have warned the user.

Your choice.